Adding test for "GET /images/(name)/history" API.

draghuram opened the pull request:
Closes #12284.

Signed-off-by: Raghuram Devarakonda draghuram@gmail.com

draghuram committed:
Adding test for "GET /images/(name)/history" API.

Closes #12284.

Signed-off-by: Raghuram Devarakonda <draghuram@gmail.com>

runcom commented on integration-cli/docker_api_images_test.go:
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
+	c.Assert(status, check.Equals, http.StatusOK)
+	c.Assert(err, check.IsNil)
+
+	var historydata []map[string]interface{}
```
Please decode in this struct https://github.com/docker/docker/blob/master/api/types/types.go#L51

runcom commented on integration-cli/docker_api_images_test.go:
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestApiImagesHistory(c *check.C) {
```
you need to add `testRequires(c, Network)` in order for this test to work on --net none build

runcom commented on integration-cli/docker_api_images_test.go:
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
+	out, err := buildImage(name, "FROM hello-world\nENV FOO bar", false)
+	if err != nil {
+		c.Fatal(err)
+	}
```
nit, you can use `c.Assert(err, check.IsNil`)

draghuram commented:
@runcom Thanks for the review. This is my first foray into go world so any suggestions to improve code are welcome. 

I have made following changes:

- Added `testRequires(c, Network)`
- Replaced explicit error checking with Asserts.
- Added an additional assertion for the returned data (in addition to its size).

runcom commented:
@draghuram  thanks!

tiborvass commented:
LGTM

estesp commented:
LGTM
